tests/sparql_functions_test.metta

```diff
@@ -0,0 +1,119 @@
+!(extend-py! motto.sparql_extension.sparql_extension)
+
+;test filters
+!(assertEqual
+    (filter (not (bound $url)))
+    "filter (!(bound($url)))")
+
+!(assertEqual
+    (filter (gt $population  15000000))
+    "filter ($population > 15000000)")
+
+!(assertEqual
+    (filter (or (contains (str $occupation) 'Actor')  (contains (str $occupation) 'Actress')))
+    "filter (contains(str($occupation), 'Actor') || contains(str($occupation), 'Actress'))")
+
+!(assertEqual
+    (filter (and (gt $date "'1968-10-1'^^xsd:date")  (lt $date "'1968-10-30'^^xsd:date")))
```

Comment:
Can't we use `>` instead of `gt` here? I see that `filter` is not declared as having `Atom` type argument. Such declaration would prevent reduction of internal arguments, which can be useful and inconvenient depending on the situation, thus, I'm not proposing it, but it may worth considering.